src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Runtime.cs

```diff
@@ -196,6 +196,7 @@ private static int BindExistingObject(object rawObj, int jsId)
                     if (!_rawToJS.TryGetValue(rawObj, out jsObject))
                     {
                         _rawToJS.Add(jsId, jsObject = new JSObject(jsId, rawObj));
+                        _rawToJS.Add(rawObj, jsObject = new JSObject(jsId, rawObj));
```

Comment:
Why not just ` _rawToJS.Add(rawObj, jsObject);` ?

Reply:
A new JSObject is created because it only enters when the value is not found.